Shared/Supporting Files/Views/SampleLink.swift

```diff
@@ -0,0 +1,129 @@
+    }
+    
+    var body: some View {
+        NavigationLink {
+            SampleDetailView(sample: sample)
+                .id(sample.name)
+        } label: {
+            SampleRow(
+                name: sample.name.boldingFirstOccurrence(of: textToBold),
+                description: sample.description.boldingFirstOccurrence(of: textToBold)
+            )
+        }
+    }
+}
+
+private extension SampleLink {
+    struct SampleRow: View {
+        /// The name string of the sample with attributes.
+        let name: AttributedString
+        
+        /// The description string of the sample with attributes.
+        let description: AttributedString
+        
+        /// A Boolean value indicating whether the sample's description is showing.
+        @State private var isShowingDescription = false
+        
+        /// The names of the favorited samples loaded from user defaults.
+        @AppStorage(UserDefaults.favoritedSamplesKey) private var favoritedNames: [String] = []
+        
+        /// A Boolean value indicating whether the sample is favorited.
+        private var isFavorited: Bool {
+            favoritedNames.contains(String(name.characters))
+        }
+        
+        var body: some View {
+            HStack {
+                VStack(alignment: .leading) {
+                    Text(name)
+                    
+                    if isShowingDescription {
+                        Text(description)
+                            .font(.caption)
+                            .opacity(0.75)
+                            .transition(.move(edge: .top).combined(with: .opacity))
+                    }
+                }
+                Spacer()
+                
+                if isFavorited {
+                    Image(systemName: "star.fill")
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(width: 12)
+                }
+                
+                Label {} icon: {
+                    Image(systemName: "info.circle")
+                        .symbolVariant(isShowingDescription ? .fill : .none)
+                        .imageScale(.medium)
+                }
+                .onTapGesture {
+                    isShowingDescription.toggle()
+                }
+            }
+            .contextMenu {
+                Button {
+                    if isFavorited {
+                        if let index = favoritedNames.firstIndex(of: String(name.characters)) {
+                            favoritedNames.remove(at: index)
+                        }
+                    } else {
+                        favoritedNames.append(String(name.characters))
+                    }
+                } label: {
+                    if isFavorited {
+                        Label("Unfavorite", systemImage: "star.slash")
+                    } else {
+                        Label("Favorite", systemImage: "star")
+                    }
```

Comment:
While it might be a bit more verbose, consider using [`SymbolVariants`](https://developer.apple.com/documentation/swiftui/symbolvariants) such as

```swift
Label("Unfavorite", systemImage: "star")
    .symbolVariant(.slash)
```

It would make it easier if we need to further tune the symbols among platforms in the future.